pyfar/dsp/dsp.py

```diff
@@ -355,3 +355,66 @@ def _cross_fade(first, second, indices):
     result = first * window_first + second * window_second
 
     return result
+
+
+def pad_zeros(signal, pad_width, mode='after'):
+    """Pad a signal with zeros in the time domain.
+    The function supports multiple padding modes, either after the last sample,
+    before the first sample, or in the middle. Note that the mode 'before'
+    results in a linear phase shift corresponding to the number of added
+    samples. The mode 'center' can be used to pad signals with a symmetry with
+    respect to the time ``t=0``.
+
+    Parameters
+    ----------
+    signal : Signal
+        The signal which is to be extended
+    pad_width : int
+        The number of samples to be padded in samples.
+    mode : str, optional
```

Comment:
For maximum flexibility, we could also allow an integer to specify after which sample the zeros are padded

Reply:
What about filling modes (e.g., 'fill_right', 'fill_center', 'fill_left') , so pad_with could be the total number of samples of the padded signal? Could be a nice feature to adjust one signal length to the length of another

Reply:
@f-brinkmann This functionality could very easily be achieved by time shifting, extending and shifting back, I think I'd prefer not to have the added complexity if a simple solution like described exists.
@sikersten I did think about this, but it's only a matter of inserting `(desired - actual)` rather than `desired`. I've chosen the version analog to numpy's `pad` function.